contracts/base-extensions/BaseOptimisticRebalanceExtension.sol

```diff
@@ -0,0 +1,390 @@
+            _newComponents,
+            _newComponentsAuctionParams,
+            _oldComponentsAuctionParams,
+            _shouldLockSetToken,
+            _rebalanceDuration,
+            _positionMultiplier
+        ));
+        ProductSettings memory settings = productSettings[setToken];
+
+        require(assertionIds[proposalHash] == bytes32(0), "Proposal already exists");
+        require(settings.rulesHash != bytes32(""), "Rules not set");
+        require(address(settings.optimisticParams.optimisticOracleV3) != address(0), "Oracle not set");
+
+
+        bytes memory claim = _constructClaim(proposalHash, settings.rulesHash);
+        uint256 totalBond = _pullBond(settings.optimisticParams);
+
+        bytes32 assertionId = settings.optimisticParams.optimisticOracleV3.assertTruth(
+            claim,
+            msg.sender,
+            address(this),
+            address(0),
+            settings.optimisticParams.liveness,
+            settings.optimisticParams.collateral,
+            totalBond,
+            settings.optimisticParams.identifier,
+            bytes32(0)
+        );
+
+        assertionIds[proposalHash] = assertionId;
+        proposedProduct[assertionId] = Proposal({
+            proposalHash: proposalHash,
+            product: setToken
+        });
+
+        emit RebalanceProposed( setToken, _quoteAsset, _oldComponents, _newComponents, _newComponentsAuctionParams, _oldComponentsAuctionParams, _shouldLockSetToken, _rebalanceDuration, _positionMultiplier);
+        emit AssertedClaim(setToken, msg.sender, settings.rulesHash, assertionId, claim);
+
+    }
+   
+    /**
+     * @dev OPERATOR ONLY: Checks that the old components array matches the current components array and then invokes the 
+     * AuctionRebalanceModuleV1 startRebalance function.
+     * 
+     * Refer to AuctionRebalanceModuleV1 for function specific restrictions.
+     *
+     * @param _quoteAsset                   ERC20 token used as the quote asset in auctions.
+     * @param _oldComponents                Addresses of existing components in the SetToken.
+     * @param _newComponents                Addresses of new components to be added.
+     * @param _newComponentsAuctionParams   AuctionExecutionParams for new components, indexed corresponding to _newComponents.
+     * @param _oldComponentsAuctionParams   AuctionExecutionParams for existing components, indexed corresponding to
+     *                                      the current component positions. Set to 0 for components being removed.
+     * @param _shouldLockSetToken           Indicates if the rebalance should lock the SetToken.
+     * @param _rebalanceDuration            Duration of the rebalance in seconds.
+     * @param _positionMultiplier           Position multiplier at the time target units were calculated.
+     */
+    function startRebalance(
+        IERC20 _quoteAsset,
+        address[] memory _oldComponents,
+        address[] memory _newComponents,
```

Comment:
let's add our optional asset allow list check on the inputs `_oldComponents` and `_newComponents` here

Reply:
I think only checking "_newComponents" should be enough though. I think we will want to be able to rebalance out of non-allowed assets. (For example when we change the set of allowed assets, or we receive a non-allowed asset as an airdrop etc.)

Reply:
oh yes i agree, checking `_newComponents` is sufficient